wasm-node/src/axum_server.rs

```diff
@@ -0,0 +1,96 @@
+use std::net::SocketAddr;
+
+use anyhow::Result;
+use axum::body;
+use axum::extract::{Path, State};
+use axum::http::{header, StatusCode};
+use axum::response::{IntoResponse, Response};
+use axum::routing::get;
+use axum::Router;
+use clap::Parser;
+use libp2p::Multiaddr;
+use rust_embed::RustEmbed;
+use serde::{Deserialize, Serialize};
+use tokio::{spawn, time};
+
+use celestia_node::network::Network;
+
+const BIND_ADDR: &str = "127.0.0.1:9876";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WasmNodeArgs {
+    pub network: Network,
+    pub bootnodes: Vec<Multiaddr>,
+}
+
+#[derive(RustEmbed)]
+#[folder = "pkg"]
+struct WasmPackage;
+
+#[derive(RustEmbed)]
+#[folder = "rsc"]
+struct StaticResources;
+
+#[derive(Debug, Clone, Parser)]
+struct Args {
+    /// Network to connect.
+    #[arg(short, long, value_enum, default_value_t)]
+    network: Network,
+
+    /// Bootnode multiaddr, including peer id. Can be used multiple times.
+    #[arg(long)]
+    bootnode: Vec<Multiaddr>,
+
+    /// Address to serve app at
+    #[arg(long, default_value = BIND_ADDR)]
+    listen_addr: SocketAddr,
+}
+
+pub async fn run() -> Result<()> {
+    let args = Args::parse();
+
+    let state = WasmNodeArgs {
+        network: args.network,
+        bootnodes: args.bootnode,
+    };
+
+    let app = Router::new()
+        .route("/", get(serve_index_html))
+        .route("/js/*script", get(serve_embedded_path::<StaticResources>))
+        .route("/wasm/*binary", get(serve_embedded_path::<WasmPackage>))
```

Comment:
why do we have those `*script` and `*binary`?

Reply:
Route without wildcard isn't interpreted as prefix, but rather as an [entire path](https://docs.rs/axum/latest/axum/routing/struct.Router.html#static-paths), so `/js/run_node.js` path wouldn't match `/js/` route here.
Ultimately I went with using `*path` as a wildcard name for both, seemed most generic and fitting, since we then we deconstruct it with axum's `Path` extractor.